zad5.py

```diff
@@ -1,49 +1,47 @@
+from math import log10
 from random import randint
-end = None
 
 
 def is_prime(n):
     if n < 2:
         return False
-    if n == 2:
+    elif n == 2:
         return True
+    elif n % 2 == 0:
+        return False
 
-    x = 2
-    while x*x <= n:
-        if n % x == 0:
+    div = 3
+    while div * div <= n:
+        if n % div == 0:
             return False
-
-        x += 2
-    end
-
+        div += 2
     return True
 
 
-def can_cut(tab):
-    a = tab[0] * 2 + tab[1]
-
-    b = 0
-    for i in tab[2:]:
-        b <<= 1
-        b |= i
-    end
+def bin_to_dec(num):
+    a = 0
+    i = 0
+    while num > 0:
+        num, rem = divmod(num, 10)
+        a += rem * (2 ** i)
+        i += 1
+    return a
```

Comment:
Taki ładny improvement
```suggestion
def bin_to_dec(num):
    a = 0
    i = 1
    while num > 0:
        num, rem = divmod(num, 10)
        a += rem * i
        i <<= 1
    return a
```